src/features/lightspeed/inlineSuggestions.ts

```diff
@@ -12,40 +12,194 @@ import {
+  documentDirPath: string;
+  documentFilePath: string;
+  documentUri: string;
+  parsedAnsibleDocument: yaml.YAMLMap[];
+}
+
+interface SuggestionMatchInfo {
+  currentLineText: vscode.TextLine;
+  lineToExtractPrompt: vscode.TextLine;
+  taskMatchedPattern: RegExpMatchArray | null;
+  spacesBeforeCursor: number;
+  spacesBeforePromptStart: number;
+  suggestionMatchType: LIGHTSPEED_SUGGESTION_TYPE | undefined;
+}
+
+interface InlinePosition {
+  document: vscode.TextDocument;
+  position: vscode.Position;
+  context: vscode.InlineCompletionContext;
+}
+
+export interface CallbackEntry {
+  (
+    suggestionDisplayed: SuggestionDisplayed,
+    inlinePosition: InlinePosition
+  ): Promise<vscode.InlineCompletionItem[]>;
+}
+
+export const onTextEditorNotActive: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  suggestionDisplayed.reset();
+  return [];
+};
+
+const onNotForMe: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  lightSpeedManager.statusBarProvider.statusBar.hide();
+  suggestionDisplayed.reset();
+  return [];
+};
+
+const onCancellationRequested: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  suggestionDisplayed.reset();
+  return [];
+};
+
+const onLightspeedIsDisabled: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  console.debug("[ansible-lightspeed] Ansible Lightspeed is disabled.");
+  lightSpeedManager.statusBarProvider.updateLightSpeedStatusbar();
+  suggestionDisplayed.reset();
+  return [];
+};
+
+const onLightspeedURLMisconfigured: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  vscode.window.showErrorMessage(
+    "Ansible Lightspeed URL is empty. Please provide a URL."
+  );
+  suggestionDisplayed.reset();
+  return [];
+};
+
+const onRetrySuggestion: CallbackEntry = async function (
+  suggestionDisplayed: SuggestionDisplayed
+) {
+  return suggestionDisplayed.cachedCompletionItem;
+};
+
+const onRefusedSuggestion: CallbackEntry = async function () {
+  vscode.commands.executeCommand(LightSpeedCommands.LIGHTSPEED_SUGGESTION_HIDE);
+  return [];
+};
+
+const onRequestInProgress: CallbackEntry = async function () {
+  lightSpeedManager.apiInstance.inlineSuggestionFeedbackSent = true;
+  vscode.commands.executeCommand(
+    LightSpeedCommands.LIGHTSPEED_SUGGESTION_HIDE,
+    UserAction.IGNORED
+  );
+  return [];
+};
+
+const onDefault: CallbackEntry = function (
+  suggestionDisplayed: SuggestionDisplayed,
+  inlinePosition: InlinePosition
+) {
+  const suggestionItems = getInlineSuggestionItems(inlinePosition);
+  return suggestionItems;
+};
+
+const CompletionState = {
+  TextEditorNotActive: onTextEditorNotActive,
+  NotForMe: onNotForMe,
+  CancellationRequested: onCancellationRequested,
+  LightspeedIsDisabled: onLightspeedIsDisabled,
+  LightspeedURLMisconfigured: onLightspeedURLMisconfigured,
+  RetrySuggestion: onRetrySuggestion,
```

Comment:
In this state we are returning the recommendation from cache, `RetrySuggestion` name can be misleading as it might suggest we are requesting Lightspeed service for a new suggestion. How about `CacheSuggestion: onCacheSuggestion`?